### ISSUE TITLE
cluster health status not working

### ISSUE THREAD

**Issue by donbowman** [tikv/pd#3077]
## Bug Report

In #2964 I commented on the cluster health status metric (pd_cluster_health_status) having an issue w/ stale members.
The resolution also corrected the inverted boolean sense (so now 1 == good, 0 == bad).

However, I now in v4.0.7 see two issues:

1. the metric is only available from one of the pd (the other pd in the cluster have no metric called `pd_cluster_health_status`)
2. the value still shows 0

so e.g;

```
curl http://pd-0:2379 |grep health_status -> no results
curl http://pd-1:2379 | grep health_status -> no results
curl http://pd-2:2379 | grep health_status
# HELP pd_cluster_health_status Status of the cluster.
# TYPE pd_cluster_health_status gauge
pd_cluster_health_status{name="db-pd-0"} 0
pd_cluster_health_status{name="db-pd-1"} 0
pd_cluster_health_status{name="db-pd-2"} 0
```

a) why the 0?
b) why does db-0 and db-1 not respond?

this is making it impossible to construct a prometheus alert

### What did you expect to see?

i expect that:
a) all pd show cluster_health_status metric
b) the result code is 1 for good, 0 for bad (or at least document it?)

### What version of PD are you using (`pd-server -V`)?

v4.0.7

**Comment by disksing** [comment]
Hi @donbowman , the metrics are only recorded by the leader, so you can only see it from PD leader. Also, the previous fix is only merged to master branch. (If you have any needs, please let us know so we can cherry-pick it to release-4.0 !)

**Comment by donbowman** [comment]
this makes it impossible to monitor w/ prometheus.
is there a reason the health stats can't be on each?

the rationale:

I have a (currently) 3 node pd cluster. It runs in kubernetes. I monitor w/ prometheus.
I want a health check so that if the reachability fails, e.g. if dns is wrong or something, i get an alert.

but, if i monitor this variable, it constantly gives false alarms, each time the leadership changes, there is an alarm since i see a threshold change.

Is there an alternative suggestion? Or this is only intended for use w/ curl and manual?

**Comment by disksing** [comment]
We usually deploy a prometheus server to pull metrics from different pd servers. External tools can monitor the state by querying prometheus, rather than querying pd directly.
There are many other benefits to using a prometheus server, such as the data is persistent and you can run more complex aggregate queries. You may want to consider this approach.
If there are other considerations that prevent you from doing so, please report back to us and we can make the necessary optimizations for this usage.

**Comment by donbowman** [comment]
yes that is what i did. I have a single prometheus, pulling all the metrics
however, there is no method to create an alert here, since every time there is a change in group leader it alerts.

E.g. this will give many false positives per day. The reason is, each pod suddenly loses the stats.

So, do you have any recommendation to stably create a rule on this field? else I don't think it works for this (or any?) use.

```
+# See https://github.com/tikv/pd/issues/3077
+#        - alert: PDHealth
+#          annotations:
+#            description: |
+#              PD {{$labels.name}} has health issues as seen by {{$labels.pod_name}}
+#            summary: PD {{$labels.name}} has health issues as seen by {{$labels.pod_name}}
+#          expr: avg_over_time(pd_cluster_health_status[10m]) < 0.9
+#          for: 20m
+#          labels:
+#            cluster: __CLUSTER__
+#            severity: warning
```

**Comment by disksing** [comment]
Usually, we only observe the metrics of the current leader node. Recording metrics on every node may not fully solve the problem here either, because if a node fails, the last reported state is maintained on the prometheus server. I wonder if the alert can be based on the last updated data.

**Comment by donbowman** [comment]
if the reachability metrics were on each server, from its vantage point, it would solve the problem.
That's what the < 0.9 is for in my alert rule.

do you have an alternative suggestion? has anyone ever tried to do alert management with prometheus with tikv?
